crates/libs/windows/src/core/factory_cache.rs

```diff
@@ -6,13 +6,18 @@ use core::sync::atomic::{AtomicPtr, Ordering};
 #[doc(hidden)]
 pub struct FactoryCache<C, I> {
     shared: AtomicPtr<core::ffi::c_void>,
+    library: Option<&'static str>,
```

Comment:
Maybe just `&'static str` and let an empty string mean the absence of a library to avoid having to check two things at load time.